evm/db/chain.py

```diff
@@ -88,36 +84,14 @@ class TransactionKey(rlp.Serializable):
 
 
 class BaseChainDB(metaclass=ABCMeta):
-    trie_class = None
-    empty_root_hash = None
-
     #
     # Trie
     #
-    def set_trie(self, trie_class: Union[Type[HexaryTrie], Type[BinaryTrie]]) -> None:
-        """
-        Sets trie_class and root_hash.
-        """
-        if trie_class is HexaryTrie:
-            empty_root_hash = BLANK_ROOT_HASH
-        elif trie_class is BinaryTrie:
-            empty_root_hash = EMPTY_SHA3
-        else:
-            raise NotImplementedError(
-                "trie_class {} is not supported.".format(trie_class)
-            )
-        self.trie_class = trie_class
-        self.empty_root_hash = empty_root_hash
```

Comment:
The root hash code feels like unnecessary inspection on the trie class details, so I'm happy to see that go. Maybe we can push the blank root hash responsibility into the trie class. IIRC, the class already keeps track of what a blank hash is anyway.

Anyway, I guess that's only relevant when we bring back configurable trie classes.

Reply:
I think this stuff is implemented in the wrong place so....

> the cost of keeping the configurable tries around might be higher than the cost of adding them back in later

I think this is accurate.